UnitsNet/QuantityFormatter.cs

```diff
@@ -39,41 +71,62 @@ public static string Format<TUnitType>(IQuantity<TUnitType> quantity, string for
         {
             formatProvider = formatProvider ?? CultureInfo.CurrentUICulture;
 
-            var number = 0;
-            var formatString = format;
-
-            if(string.IsNullOrEmpty(formatString))
-                formatString = "g";
+            if(string.IsNullOrEmpty(format))
+                format = "g";
 
-            if(formatString.StartsWith("a") || formatString.StartsWith("s"))
+            if(UnitsNetFormatSpecifiers.Any(c => c == format[0]))
             {
-                if(formatString.Length > 1 && !int.TryParse(formatString.Substring(1), out number))
-                    throw new FormatException($"The {format} format string is not supported.");
+                // UnitsNet custom format string
 
-                formatString = formatString.Substring(0, 1);
-            }
+                int precisionSpecifier = 0;
+                char formatSpecifier = format[0];
```

Comment:
Nitpick, can reuse this variable in the above if.

Reply:
👍